### PR TITLE
Use cibuildwheel to build wheels

### PR DESCRIPTION
**Pull request by dstansby**
This reduces the amount of custom GH actions script needed to build & upload wheels.

### CODE REVIEW

**Comment by adamltyson** [comment]
Is there any logic for putting this into https://github.com/brainglobe/actions in case we need it for other projects?

**Comment by dstansby** [comment]
Sounds like a good idea, I'll take a look 👍

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -23,90 +23,33 @@ jobs:
...
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
     needs: test
-    if: contains(github.ref, 'tags')
-    runs-on: windows-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        os: [ubuntu-20.04, windows-2019, macos-10.15]
```
Comment (paddyroddy) [inline .github/workflows/test_and_deploy.yml:43]: Catalina i.e. `10.15` is pretty old at this point?

**Review by paddyroddy — APPROVED** [review]
Nice, looks good

**Pull request merged**